packages/xo-server/src/xo-mixins/xen-servers.js

```diff
@@ -425,10 +425,13 @@ export default class {
         xapi.xo.uninstall()
         delete this._xapis[server.id]
         delete this._serverIdsByPool[poolId]
+        this._xo.emit('server:disconnected', { server, xapi })
       })
+      this._xo.emit('server:connected', { server, xapi })
     } catch (error) {
       delete this._xapis[server.id]
       xapi.disconnect()::ignoreErrors()
+      this._xo.emit('server:disconnected', { server, xapi })
```

Comment:
Does it make sense to emit `disconnected` if `connected` hasn't been emitted? Or should there be a 3rd event?

Reply:
Hmm, I only need the two in my usage. But if you think it's necessary to differentiate, it can be done.

Reply:
@benjamreis I don't feel like you've answered @pdonias' question: what's the use case for emitting `disconnected` without a prior `connected`?

Reply:
Hmmm it seems I didn't understand the comment... Now that I have, I think does not make sense to emit `disconnected` indeed.